_integration-tests/start.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023-present Datadog, Inc.
+
+####################################################################################################
+# Usage: /start.sh <test> [args...]
+#
+# This program starts the service handler specified as `<test>` from the `/.bin` directory. It
+# starts a test agent session before starting the handler, and collects the traces from the session
+# upon exit and puts them at `/output/traces.json`. The handlers SDTOUT and STDERR streams are
+# redirected to `/output/stdout.log` and `/output/stderr.log` respectively.
+####################################################################################################
+
+set -euo pipefail
+
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+if [[ $# -lt 1 ]]; then
+  echo "Usage: ${SCRIPTDIR}/start.sh <test> [args...]"
+  echo "Error: missing argument value for <test>"
+  exit 2
+fi
+
+TEST_NAME="${1}"
+shift
+
+token="${TEST_NAME}-$(date +'%s.%N')"
+
+finish() {
+  rv=$?
+  local token=$1
+  echo "Service exited with status ${rv}"
+  echo "Reading traces from the agent..."
+  for (( i=0; i<5; i++ )); do
+    curl -f "http://${DD_AGENT_HOST:-localhost}:8126/test/session/traces?test_session_token=${token}" \
+      -o "/output/traces.json"
+    if [[ "$(jq '. | length' < /output/traces.json)" == "0" ]]; then
+      echo "Agent returned no traces... retrying in 1 second..."
+      sleep 1
+    else
+      break
+    fi
+  done
```

Comment:
This loop is to avoid the tests being flakey... Turns out even without significant load, the fake agent may take a short moment to reach full consistency...